pkg/apis/rke.cattle.io/v1/plan/plan.go

```diff
@@ -44,15 +53,25 @@ type Instruction struct {
 	SaveOutput bool     `json:"saveOutput,omitempty"`
 }
 
+type PeriodicInstruction struct {
+	Name          string   `json:"name,omitempty"`
+	Image         string   `json:"image,omitempty"`
+	Env           []string `json:"env,omitempty"`
+	Args          []string `json:"args,omitempty"`
+	Command       string   `json:"command,omitempty"`
```

Comment:
Looks like `PeriodicInstruction` is mostly the same as `OneTimeInstruction` with the addition of `PeriodSeconds` - which makes sense. Could we use composition here instead?
We don't have `SaveOutput` in `PeriodicInstruction`, but I'm not seeing any usages of it in `rancher/rancher`. Is it used elsewhere? I guess we don't want to remove it either way, but it's still weird then we don't have it on `PeriodicInstruction`.

Reply:
No, for whatever reason I'm unable to get this to work properly when using generated CRDs.

I do this on the system-agent side.

Reply:
A periodic instruction has an implicit `SaveOutput: true` hence why it's not necessary.